samples/extensions/dynamic_rendering/dynamic_rendering.cpp

```diff
@@ -372,6 +372,10 @@ void DynamicRendering::build_command_buffers()
 		{
 			vkb::image_layout_transition(draw_cmd_buffer,
 			                             swapchain_buffers[i].image,
+			                             VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,
```

Comment:
Assuming the acquisition of the swapchain image is protected by a fence guarding it from the prior present, I don't see why this wouldn't be `VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT` (or better yet `VK_PIPELINE_STAGE_2_NONE`), especially as the source layout is "undefined", meaning we don't care about the contents of this image anymore.

Reply:
From that issue I mentioned seems that changing layout will happen earlier than waiting for rasterization write to image.

I'm not that much experience Vulkan engineer, but this sentence seems reasonable:
`There should be additional execution dependency with COLOR_ATTACHMENT_OUTPUT stage, otherwise image transition can start before image acquire operation finished. The destination stage is set to COLOR_ATTACHMENT_OUTPUT and this guarantees that writes from rasterization will wait. But the image layout transition happens earlier, and by setting source stage to COLOR_ATTACHMENT_OUTPUT we create execution dependency with semaphore wait operation (we chain with pWaitDstStageMask which also specifies COLOR_ATTACHMENT_OUTPUT ).`

Reply:
I'll have to defer to someone with a better understanding of this than myself then.  I don't quite get why the dstStage isn't sufficient to ensure the proper ordering.